geopandas/io/file.py

```diff
@@ -34,7 +34,15 @@ def _is_url(url):
         return False
 
 
-def read_file(filename, bbox=None, mask=None, rows=None, **kwargs):
+def read_file(
+    filename,
+    bbox=None,
+    mask=None,
+    rows=None,
+    columns=None,
+    ignore_geometry=False,
```

Comment:
@snowman2 thanks for adding ability to omit geometry.  What I originally meant was that `columns` could be `['geometry', 'foo', 'bar']` to _include_ the geometry, or `['foo', 'bar']` to ignore the geometry.  That way the `columns` has parity with the columns of the resultant data frame - both conceptually and in terms of order.  That seems better than having `geometry` managed outside of `columns`.

I instead meant to leverage the `ignore_geometry` parameter when reading from Fiona:
```
ignore_geometry = columns is not None  'geometry' not in columns
with reader(path_or_bytes, ignore_geometry=ignore_geometry, **kwargs) as features:
```

I also suggested using `ignore_fields` at this level too, which I think you could get from:
```

if columns is not None:
    with reader(path_or_bytes, **kwargs) as features: 
        ignore_fields = set(list(features.meta["schema"]["properties"])).difference(columns)
        if ignore_fields:
            kwargs['ignore_fields'] = ignore_fields

with reader(path_or_bytes, **kwargs) as features: 
   ...as before...
```

I'm not sure how to do it without the double-open of the reader since `ignore_fields` is at that level, yet we need the list of properties in order to know what to ignore!

I'm not sure of the impact if the reader is against a URL data source instead of a file.

Reply:
> What I originally meant was that columns could be ['geometry', 'foo', 'bar'] to include the geometry, or ['foo', 'bar'] to ignore the geometry. That way the columns has parity with the columns of the resultant data frame - both conceptually and in terms of order. That seems better than having geometry managed outside of columns.

I think most use cases (100% of mine anyway) include the `geometry` column. So, from that perspective, I think not having to include it in the `columns` list would be much more convenient for most users. Then, the small percentage of the time when you actually want to exclude the `geometry` column, you can do so with the flag. This also makes the logic much simpler on the implementation side.

Reply:
Also, it matches the behavior of `Fiona`, so that adds consistency there.

Reply:
Your idea for `ignore_fields` is something I thought about originally. Opening the file twice made me hesitate. I think I will need to investigate that a bit further, try it out, and think on it.